lms/djangoapps/ccx/api/v0/views.py

```diff
@@ -156,9 +158,47 @@ def get_valid_input(request_data, ignore_missing=False):
             field_errors['max_students_allowed'] = {'error_code': 'invalid_max_students_allowed'}
     elif 'max_students_allowed' in request_data:
         field_errors['max_students_allowed'] = {'error_code': 'null_field_max_students_allowed'}
+
+    course_modules = request_data.get('course_modules')
+    if course_modules is not None:
+        if isinstance(course_modules, list):
+            for course_module_id in course_modules:
+                try:
+                    UsageKey.from_string(course_module_id)
+                except InvalidKeyError:
+                    field_errors['course_modules'] = {'error_code': 'invalid_course_module_keys'}
+                    break
+            else:
+                valid_input['course_modules'] = course_modules
```

Comment:
The passed-in course modules should be de-duplicated somewhere - perhaps here. (I assume there's no reason to have more then one of the same module in the list of course modules.) And a test should be added to ensure that duplicates are correctly discarded.